cmd/bnscli/clitests/batch.test

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+set -e
+
+msgs=`tempfile`
+
+# Combine all messages in a single file to build a batch transaction.
+bnscli send-tokens -src 'seq:foo/src/1' -dst "seq:foo/dst/1" -amount "2 IOV" -memo "sending 2 IOV" >> $msgs
+bnscli send-tokens -src 'seq:foo/src/2' -dst "seq:foo/dst/2" -amount "9 DOGE" -memo "sending 9 DOGE" >> $msgs
+bnscli send-tokens -src 'seq:foo/src/3' -dst "seq:foo/dst/3" -amount "7 BTC" -memo "sending 7 BTC" >> $msgs
```

Comment:
This is very cool how you can just append them.
I guess you took advantage of the protobuf repeated trick?

Reply:
No, protobuf cannot be used for streaming on it's own. I had to reimplement how each transaction is [written and read](https://github.com/iov-one/weave/blob/efe6ed806da868fc8fc8c46131bb0af4930c2c73/cmd/bnscli/common.go#L18-L60) to include the message size information.

https://developers.google.com/protocol-buffers/docs/techniques#streaming